rust/index/src/hnsw_provider.rs

```diff
@@ -173,62 +174,30 @@ impl HnswIndexProvider {
             let file_path = index_storage_path.join(file);
             // For now, we never evict from the cache, so if the index is being loaded, the file does not exist
             let file_handle = tokio::fs::File::create(&file_path).await;
-            let file_handle = match file_handle {
+            let mut file_handle = match file_handle {
                 Ok(file) => file,
                 Err(e) => {
                     tracing::error!("Failed to create file: {}", e);
                     return Err(Box::new(HnswIndexProviderFileError::IOError(e)));
                 }
             };
-            let total_bytes_written = self
-                .copy_stream_to_local_file(reader, file_handle)
-                .instrument(tracing::info_span!(parent: Span::current(), "hnsw provider file read", file = file))
-                .await?;
-            tracing::info!(
-                "Copied {} bytes from storage key: {} to file: {}",
-                total_bytes_written,
-                key,
-                file_path.to_str().unwrap()
-            );
-            // bytes is an AsyncBufRead, so we fil and consume it to a file
-            tracing::info!("Loaded hnsw index file: {}", file);
-        }
-        Ok(())
-    }
-
-    async fn copy_stream_to_local_file(
-        &self,
-        stream: Box<dyn stream::Stream<Item = ByteStreamItem> + Unpin + Send>,
-        file_handle: tokio::fs::File,
-    ) -> Result<u64, Box<HnswIndexProviderFileError>> {
-        let mut total_bytes_written = 0;
-        let mut file_handle = file_handle;
-        let mut stream = stream;
-        while let Some(res) = stream.next().await {
-            let chunk = match res {
-                Ok(chunk) => chunk,
-                Err(e) => {
-                    return Err(Box::new(HnswIndexProviderFileError::StorageGetError(e)));
-                }
-            };
-
-            let res = file_handle.write_all(&chunk).await;
+            let res = file_handle.write_all(&buf).await;
```

Comment:
i think this will end up being slower now that we have this get() copy in the middle FYI

Reply:
Discussed offline that we are going to do the 4MB split now so this will end up mattering very less